homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,260 @@
+        name = None
+        _LOGGER.info("Discovered a Daikin AC on %s", host)
+    else:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        _LOGGER.info("Added Daikin AC on %s", host)
+
+    api = daikin_api_setup(hass, host, name)
+    add_devices([DaikinClimate(api)], True)
+
+
+class DaikinClimate(ClimateDevice):
+    """Representation of a Daikin HVAC."""
+
+    def __init__(self, api):
+        """Initialize the climate device."""
+        from pydaikin import appliance
+
+        self._api = api
+        self._force_refresh = False
+        self._list = {
+            ATTR_OPERATION_MODE: list(
+                map(str.title, set(HA_STATE_TO_DAIKIN.values()))
+            ),
+            ATTR_FAN_MODE: list(
+                map(
+                    str.title,
+                    appliance.daikin_values(HA_ATTR_TO_DAIKIN[ATTR_FAN_MODE])
+                )
+            ),
+            ATTR_SWING_MODE: list(
+                map(
+                    str.title,
+                    appliance.daikin_values(HA_ATTR_TO_DAIKIN[ATTR_SWING_MODE])
+                )
+            ),
+        }
+
+    def get(self, key):
+        """Retrieve device settings from API library cache."""
+        value = None
+        cast_to_float = False
+
+        if key in [ATTR_TEMPERATURE, ATTR_INSIDE_TEMPERATURE,
+                   ATTR_CURRENT_TEMPERATURE]:
+            value = self._api.device.values.get('htemp')
+            cast_to_float = True
+        if key == ATTR_TARGET_TEMPERATURE:
+            value = self._api.device.values.get('stemp')
+            cast_to_float = True
+        elif key == ATTR_OUTSIDE_TEMPERATURE:
+            value = self._api.device.values.get('otemp')
+            cast_to_float = True
+        elif key == ATTR_FAN_MODE:
+            value = self._api.device.represent('f_rate')[1].title()
+        elif key == ATTR_SWING_MODE:
+            value = self._api.device.represent('f_dir')[1].title()
+        elif key == ATTR_OPERATION_MODE:
+            # Daikin can return also internal states auto-1 or auto-7
+            # and we need to translate them as AUTO
+            value = re.sub(
+                '[^a-z]',
+                '',
+                self._api.device.represent('mode')[1]
+            ).title()
+
+        if value is None:
+            _LOGGER.warning("Invalid value requested for key %s", key)
+        else:
+            if value == "-" or value == "--":
+                value = None
+            elif cast_to_float:
+                try:
+                    value = float(value)
+                except ValueError:
+                    value = None
+
+        return value
+
+    def set(self, settings):
+        """Set device settings using API."""
+        values = {}
+
+        for attr in [ATTR_TEMPERATURE, ATTR_FAN_MODE, ATTR_SWING_MODE,
+                     ATTR_OPERATION_MODE]:
+            if attr in settings and settings[attr] is not None:
```

Comment:
```py
value = settings.get(attr)
if value is None:
    continue
daikin_attr = ...
```

Reply:
I know for sure setting[attr] is not None so I don't need to guard safe it and add an extra if, I verify it above; so I don't see any benefit (performance, code readability, u name it) to add this change.
Can you please explain your suggestion @MartinHjelmare or you just want me to split the condition from above ?